lib/node_modules/@stdlib/random/base/xorshift/src/main.c

```diff
@@ -0,0 +1,424 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/random/base/xorshift.h"
+#include "stdlib/random/base/shared.h"
+#include <stdlib.h>
+#include <stdint.h>
+#include <string.h>
+
+// Forward declarations:
+static inline int8_t next( struct BasePRNGObject *obj, uint64_t *out );
+static inline int8_t normalized( struct BasePRNGObject *obj, double *out );
+static inline void xorshift_free( struct BasePRNGObject *obj );
+
+// Define the maximum unsigned 32-bit integer: 4294967295 => 0xffffffff => 11111111111111111111111111111111
+static const uint32_t MAX_UINT32 = 0xffffffff;
+
+// Define the normalization constant:
+static const double NORMALIZATION_CONSTANT = (double)MAX_UINT32;
+
+/**
+* XORSHIFT PRNG.
+*
+*/
+static const struct BasePRNG xorshift_prng = {
```

Comment:
```suggestion
static const struct BasePRNG xorshift32_prng = {
```

and elsewhere after renaming the package to `xorshift32`.